src/cache.c

```diff
@@ -30,20 +29,24 @@ struct hlist_node {
 
 typedef struct {
     void *value;
+    bool alive; /* whether this cache is alive or a history in hash map */
     uint32_t key;
     uint32_t frequency;
     struct list_head list;
-    struct hlist_node ht_list;
-} lfu_entry_t;
+    struct hlist_node
+        ht_list; /* it contains both live and history of evicted cache */
+} cache_entry_t;
 
 typedef struct {
     struct hlist_head *ht_list_head;
 } hashtable_t;
 
 typedef struct cache {
-    struct list_head *lists[THRESHOLD];
-    uint32_t list_size;
-    hashtable_t *map;
+    struct list_head list;
+    struct list_head ghost_list;
```

Comment:
Document the use of ghost_list, optionally referring adaptive replacement cache (ARC).